tests/returntypes/test_jax_qnode_new.py

```diff
@@ -2190,3 +2190,1194 @@ def circuit(x):
 
         assert isinstance(hess[1], jax.numpy.ndarray)
         assert hess[1].shape == (4, 2, 2)
+
+
+shots = [100, (1, 20, 100), (1, (20, 1), 100), (1, (5, 4), 100)]
```

Comment:
If these are shots meant to test the shot vector case, should `100` be omitted? Having a single finite shot element seems to make tests more complicated.

Reply:
It was never really tested with shots, so I added it to double check.

Reply:
Adding this case is great. What I'm getting at here is more so organizational: I think finite shots should rather go with the `shots=None` (also potentially changing the tolerance value) because the shapes of those two cases are the same.

Another note: I think it would be good to extract the shot vector tests into their own files: this file is already 2k lines long and ~1.3k lines are being added there. Best to separate them.

Out of these two suggestions, I think the second is more burning, the first would be nice to have, but may not be so worth it when considering the ratio of work required/value brought.